recipes/yder/all/patches/1.4.18-0001-shared-static-conan.patch

```diff
@@ -0,0 +1,406 @@
++            VERSION "${LIBRARY_VERSION}"
++            SOVERSION "${LIBRARY_SOVERSION}"
++            WINDOWS_EXPORT_ALL_SYMBOLS TRUE
++            EXPORT_NAME Yder)
++    if (WIN32)
++        set_target_properties(yder PROPERTIES
++                SUFFIX "-${LIBRARY_VERSION_MAJOR}.dll")
++    endif ()
++    if (NOT MSVC)
++        target_compile_options(yder PRIVATE -Wextra)
++    endif ()
++    set(yder_lib yder)
++endif()
++
+ option(DOWNLOAD_DEPENDENCIES "Download required dependencies" ON)
+ 
+ option(SEARCH_ORCANIA "Search for Orcania library" ON)
+ if (SEARCH_ORCANIA)
+-    set(Orcania_FIND_QUIETLY ON) # force to find Orcania quietly
+-    include(FindOrcania)
+     find_package(Orcania ${ORCANIA_VERSION_REQUIRED} QUIET) # try to find orcania
+-    if (NOT ORCANIA_FOUND)
++    if (NOT Orcania_FOUND)
+         if (DOWNLOAD_DEPENDENCIES)
+             include(DownloadProject)
+             download_project(PROJ orcania # ... otherwise, download archive
+-                URL "https://github.com/babelouest/orcania/archive/v${ORCANIA_VERSION_REQUIRED}.tar.gz"
+-                QUIET)
++                             URL "https://github.com/babelouest/orcania/archive/v${ORCANIA_VERSION_REQUIRED}.tar.gz"
++                             QUIET)
+             add_subdirectory(${orcania_SOURCE_DIR} ${orcania_BINARY_DIR})
+-            include_directories(${orcania_SOURCE_DIR}/include)
+-            include_directories(${orcania_BINARY_DIR})
+-            add_dependencies(yder orcania)
+-            set(ORCANIA_LIBRARIES orcania)
+-            set(LIBS ${LIBS} ${ORCANIA_LIBRARIES})
++            if (NOT TARGET Orcania::Orcania)
++                add_library(Orcania::Orcania ALIAS orcania)
++            endif ()
++            if (NOT TARGET Orcania::Orcania-static AND TARGET orcania_static)
++                add_library(Orcania::Orcania-static ALIAS orcania_static)
++            endif ()
+         else ()
+             message( FATAL_ERROR "Orcania not found")
+         endif ()
+     else()
+-        message(STATUS "Orcania found")
+-        set(LIBS ${LIBS} ${ORCANIA_LIBRARIES})
+-        include_directories(${ORCANIA_INCLUDE_DIRS})
+-        include_directories(${orcania_BINARY_DIR})
++        if ("${ORCANIA_VERSION_STRING}" VERSION_GREATER_EQUAL "${ORCANIA_VERSION_REQUIRED}")
++            message(STATUS "Orcania found: ${ORCANIA_VERSION_STRING}")
++        else ()
++            message( FATAL_ERROR "Orcania version required: ${ORCANIA_VERSION_REQUIRED} - version installed: ${ORCANIA_VERSION_STRING}")
++        endif ()
++    endif ()
++else ()
++    if (NOT TARGET Orcania:: Orcania)
++        add_library(Orcania::Orcania IMPORTED UNKNOWN)
++        set_target_properties(Orcania::Orcania PROPERTIES IMPORTED_LOCATION "orcania")
+     endif ()
+ endif ()
+ 
+-target_link_libraries(yder ${LIBS} ${ORCANIA_LIBRARIES} ${SYSTEMD_LIBRARIES})
++if (BUILD_SHARED)
++    target_link_libraries(yder PRIVATE $<TARGET_NAME:Orcania::Orcania> ${SYSTEMD_LIBRARIES})
```

Comment:
Does this mean that a shared build of `yder` can't link to a static `orcania` package?

Reply:
Not with the cmake script unmodified.
It is patched in `_patch_sources()`.

Reply:
Gotcha.

Reply:
When/if https://github.com/conan-io/conan/issues/12367 lands, the `replace_in_file` can be replaced with that.